orderer/common/follower/follower_chain.go

```diff
@@ -354,7 +357,14 @@ func (c *Chain) pull() error {
 	// Trigger creation of a new consensus.Chain.
 	c.logger.Info("Block pulling finished successfully, going to switch from follower to a consensus.Chain")
 	c.halt()
-	c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID())
+	for {
+		if isExec := c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID()); isExec ||
+			c.removingExternal.Load() == 1 {
+			break
+		}
```

Comment:
This section requires an explanation and comments to clarify what is happening here...

Reply:
done.